### DISCUSSION TITLE
[Idea] Prestashop Filesystem Cache

### DISCUSSION THREAD

**Discussion by Dreimus** [PrestaShop/PrestaShop#35643] [Ideas]
### The problem

Currently if you need Prestashop Cache, you have to possibilities to use :
* memcache
* memcached
* apcu
* xcache

Why a filesystem cache couldn't be a "basic" option ?

### The solution

Implement an Filesystem cache option that register file serialization in `var/cache/<env>` folder

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
We've had one in the past, and it was terrible. It produced many files, and it was almost impossible to invalidate it correctly. We could utilize Symfony cache implementation.

Overall, it's a good idea, but this is a more significant project with many decisions to make.

If you are motivated to help with it, I suggest waiting for @PrestaShop/tech-council and @PrestaShop/product-council to see if this project could be somehow prioritized with reviews and project members' availability. Otherwise, it would be a waste of your time.

**Comment by DracoBluePL** [comment]
Where is Redis? Just some old cache systems.

**Comment by joelfrancis1122** [comment]
yah great

**Comment by DracoBluePL** [comment]
In modern commercial engines, typically you have:
1. File system
2. Redis
